ui/apps/platform/src/Containers/Clusters/ClustersTable.tsx

```diff
@@ -0,0 +1,246 @@
+import React, { useState } from 'react';
+import { Link } from 'react-router-dom';
+import { Flex } from '@patternfly/react-core';
+import {
+    ActionsColumn,
+    ExpandableRowContent,
+    Table,
+    Tbody,
+    Td,
+    Th,
+    Thead,
+    Tr,
+} from '@patternfly/react-table';
+
+import TbodyUnified from 'Components/TableStateTemplates/TbodyUnified';
+import useMetadata from 'hooks/useMetadata';
+import { Cluster } from 'types/cluster.proto';
+import { ClusterIdToRetentionInfo } from 'types/clusterService.proto';
+import { TableUIState } from 'utils/getTableUIState';
+
+import { formatCloudProvider } from './cluster.helpers';
+import { CertExpiryStatus, ClusterHealthStatus } from './clusterTypes';
+import ClusterDeletion from './Components/ClusterDeletion';
+import ClusterStatus from './Components/ClusterStatus';
+import CredentialExpiration from './Components/CredentialExpiration';
+import HelmIndicator from './Components/HelmIndicator';
+import OperatorIndicator from './Components/OperatorIndicator';
+import SensorUpgrade from './Components/SensorUpgrade';
+
+export type ClustersTableProps = {
+    clusterIdToRetentionInfo: ClusterIdToRetentionInfo;
+    tableState: TableUIState<Cluster>;
+    selectedClusterIds: string[];
+    onClearFilters: () => void;
+    onDeleteCluster: (cluster: Cluster) => (event: React.MouseEvent) => void;
+    toggleAllClusters: () => void;
+    toggleCluster: (clusterId) => void;
+};
+
+export const COL = {
```

Comment:
nit: `COL` seems a bit too abbreviated. Based on the code, `EXPANDABLE_COLUMNS` (or `EXPANDABLE_COLUMN_NAMES`) might be clearer and easier to understand. This could apply to the `ColId` type too. I would probably call that `ExpandableColumn` (or `ExpandableColumnName`).

Reply:
Brad and I discussed potential alternatives so **Sensor upgrade status** might not need expansion.